In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn
import numpy as np

Imported libraries together to be more streamlined and clean, same as the first cleanup notebook.

In [2]:
df = pd.read_csv('../Data/teen_phone_addiction_dataset.csv')
df.head()

,ID,Name,Age,Gender,Location,School_Grade,Daily_Usage_Hours,Sleep_Hours,Academic_Performance,Social_Interactions,...,Screen_Time_Before_Bed,Phone_Checks_Per_Day,Apps_Used_Daily,Time_on_Social_Media,Time_on_Gaming,Time_on_Education,Phone_Usage_Purpose,Family_Communication,Weekend_Usage_Hours,Addiction_Level
0,1,Shannon Francis,13,Female,Hansonfort,9th,4.0,6.1,78,5,...,1.4,86,19,3.6,1.7,1.2,Browsing,4,8.7,10.0
1,2,Scott Rodriguez,17,Female,Theodorefort,7th,5.5,6.5,70,5,...,0.9,96,9,1.1,4.0,1.8,Browsing,2,5.3,10.0
2,3,Adrian Knox,13,Other,Lindseystad,11th,5.8,5.5,93,8,...,0.5,137,8,0.3,1.5,0.4,Education,6,5.7,9.2
3,4,Brittany Hamilton,18,Female,West Anthony,12th,3.1,3.9,78,8,...,1.4,128,7,3.1,1.6,0.8,Social Media,8,3.0,9.8
4,5,Steven Smith,14,Other,Port Lindsaystad,9th,2.5,6.7,56,4,...,1.0,96,20,2.6,0.9,1.1,Gaming,10,3.7,8.6


Running basic head, info, value counts, and shape to get information on what to do first for cleanup.

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   ID                      3000 non-null   int64  
 1   Name                    3000 non-null   str    
 2   Age                     3000 non-null   int64  
 3   Gender                  3000 non-null   str    
 4   Location                3000 non-null   str    
 5   School_Grade            3000 non-null   str    
 6   Daily_Usage_Hours       3000 non-null   float64
 7   Sleep_Hours             3000 non-null   float64
 8   Academic_Performance    3000 non-null   int64  
 9   Social_Interactions     3000 non-null   int64  
 10  Exercise_Hours          3000 non-null   float64
 11  Anxiety_Level           3000 non-null   int64  
 12  Depression_Level        3000 non-null   int64  
 13  Self_Esteem             3000 non-null   int64  
 14  Parental_Control        3000 non-null   int64  
 15

3000 rows, 25 columns, no missing values in any column. Bigger dataset than Teen_Mental_Health, but same clean starting point, no nulls to fix.

In [4]:
df.shape

(3000, 25)

In [5]:
df.describe()

,ID,Age,Daily_Usage_Hours,Sleep_Hours,Academic_Performance,Social_Interactions,Exercise_Hours,Anxiety_Level,Depression_Level,Self_Esteem,Parental_Control,Screen_Time_Before_Bed,Phone_Checks_Per_Day,Apps_Used_Daily,Time_on_Social_Media,Time_on_Gaming,Time_on_Education,Family_Communication,Weekend_Usage_Hours,Addiction_Level
count,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000
mean,1500.500000,15.969667,5.020667,6.489767,74.947333,5.097667,1.040667,5.590000,5.460333,5.546333,0.507333,1.006733,83.093000,12.609333,2.499233,1.525267,1.016333,5.459667,6.015100,8.881900
std,866.169729,1.989489,1.956501,1.490713,14.684156,3.139333,0.734620,2.890678,2.871557,2.860754,0.500030,0.492878,37.747044,4.611486,0.988201,0.932701,0.648341,2.864572,2.014776,1.609598
min,1.000000,13.000000,0.000000,3.000000,50.000000,0.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000,20.000000,5.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000
25%,750.750000,14.000000,3.700000,5.500000,62.000000,2.000000,0.500000,3.000000,3.000000,3.000000,0.000000,0.700000,51.000000,9.000000,1.800000,0.800000,0.500000,3.000000,4.700000,8.000000
50%,1500.500000,16.000000,5.000000,6.500000,75.000000,5.000000,1.000000,6.000000,5.000000,6.000000,1.000000,1.000000,82.000000,13.000000,2.500000,1.500000,1.000000,5.000000,6.000000,10.000000
75%,2250.250000,18.000000,6.400000,7.500000,88.000000,8.000000,1.500000,8.000000,8.000000,8.000000,1.000000,1.400000,115.250000,17.000000,3.200000,2.200000,1.500000,8.000000,7.400000,10.000000
max,3000.000000,19.000000,11.500000,10.000000,100.000000,10.000000,4.000000,10.000000,10.000000,10.000000,1.000000,2.600000,150.000000,20.000000,5.000000,4.000000,3.000000,10.000000,14.000000,10.000000


Age runs 13 to 19, matching the first dataset. Academic_Performance here is on a 0 to 100 scale, not the 2.00 to 4.00 GPA scale used in the first dataset. Keeping that in mind for later, the two fields measure the same idea but are not directly comparable without converting one to match the other. Addiction_Level goes up to 10.0 and the mean sits close to 9, higher than the 25th percentile suggests, worth a closer look.

In [6]:
df.duplicated().sum()

np.int64(0)

Zero duplicate rows.

In [7]:
df['ID'].is_unique

True

ID column is unique for every row, good candidate for a primary key once renamed.

In [8]:
df['Name'].duplicated().sum()

np.int64(67)

67 repeated names out of 3000 rows. Expected with randomly generated synthetic names, and Name is not useful for analysis anyway, so it gets dropped along with ID's raw form once user_id is built.

In [9]:
df['Location'].nunique()

2726

2726 unique values out of 3000 rows. Location is close to a free text field here, one value per person with almost no repeats, so it will not group into anything meaningful. Dropping it, same reasoning as dropping Name.

In [10]:
df.drop(columns=['Name', 'Location'], inplace=True)
df.head()

,ID,Age,Gender,School_Grade,Daily_Usage_Hours,Sleep_Hours,Academic_Performance,Social_Interactions,Exercise_Hours,Anxiety_Level,...,Screen_Time_Before_Bed,Phone_Checks_Per_Day,Apps_Used_Daily,Time_on_Social_Media,Time_on_Gaming,Time_on_Education,Phone_Usage_Purpose,Family_Communication,Weekend_Usage_Hours,Addiction_Level
0,1,13,Female,9th,4.0,6.1,78,5,0.1,10,...,1.4,86,19,3.6,1.7,1.2,Browsing,4,8.7,10.0
1,2,17,Female,7th,5.5,6.5,70,5,0.0,3,...,0.9,96,9,1.1,4.0,1.8,Browsing,2,5.3,10.0
2,3,13,Other,11th,5.8,5.5,93,8,0.8,2,...,0.5,137,8,0.3,1.5,0.4,Education,6,5.7,9.2
3,4,18,Female,12th,3.1,3.9,78,8,1.6,9,...,1.4,128,7,3.1,1.6,0.8,Social Media,8,3.0,9.8
4,5,14,Other,9th,2.5,6.7,56,4,1.1,1,...,1.0,96,20,2.6,0.9,1.1,Gaming,10,3.7,8.6


Dropped Name and Location. Neither adds analytical value and both were closer to noise than a real category.

In [11]:
df['Gender'].value_counts()

Gender
Male      1016
Female    1007
Other      977
Name: count, dtype: int64

Male, Female, and Other are all close to 1000 rows each, an even three-way split. No fix needed here.

In [12]:
df['School_Grade'].value_counts()

School_Grade
12th    529
9th     526
7th     497
10th    487
8th     482
11th    479
Name: count, dtype: int64

7th through 12th grade, each with roughly 480 to 530 students. Even spread, no missing grade levels.

In [13]:
df['Phone_Usage_Purpose'].value_counts()

Phone_Usage_Purpose
Browsing        627
Other           622
Education       602
Social Media    575
Gaming          574
Name: count, dtype: int64

Five categories: Browsing, Other, Education, Social Media, Gaming. All within about 50 rows of each other.

In [14]:
df['Parental_Control'].value_counts()

Parental_Control
1    1522
0    1478
Name: count, dtype: int64

Binary field, 0 or 1, close to a 50/50 split.

In [15]:
df['Addiction_Level'].value_counts().sort_index().tail(10)

Addiction_Level
9.1       35
9.2       34
9.3       43
9.4       44
9.5       44
9.6       46
9.7       37
9.8       28
9.9       50
10.0    1524
Name: count, dtype: int64

This is the real finding in this dataset. 1524 of 3000 rows, just over half, sit at exactly 10.0, the maximum possible addiction score. The rest spread out across 1.0 to 9.9. That is a heavy ceiling effect, not an even synthetic spread like most of the other fields. Worth flagging clearly before this column drives any conclusions later.

In [16]:
df.isna().sum()

ID                        0
Age                       0
Gender                    0
School_Grade              0
Daily_Usage_Hours         0
Sleep_Hours               0
Academic_Performance      0
Social_Interactions       0
Exercise_Hours            0
Anxiety_Level             0
Depression_Level          0
Self_Esteem               0
Parental_Control          0
Screen_Time_Before_Bed    0
Phone_Checks_Per_Day      0
Apps_Used_Daily           0
Time_on_Social_Media      0
Time_on_Gaming            0
Time_on_Education         0
Phone_Usage_Purpose       0
Family_Communication      0
Weekend_Usage_Hours       0
Addiction_Level           0
dtype: int64

Confirms no missing values remain after the column drops.

Renaming columns to snake_case so field names line up with the first dataset where the same idea is being measured, matching the intent already started in the trial notebook.

In [17]:
df.rename(columns={
    'Age': 'age',
    'Gender': 'gender',
    'School_Grade': 'school_grade',
    'Daily_Usage_Hours': 'daily_usage_hours',
    'Sleep_Hours': 'sleep_hours',
    'Academic_Performance': 'academic_performance',
    'Social_Interactions': 'social_interactions',
    'Exercise_Hours': 'exercise_hours',
    'Anxiety_Level': 'anxiety_level',
    'Depression_Level': 'depression_level',
    'Self_Esteem': 'self_esteem',
    'Parental_Control': 'parental_control',
    'Screen_Time_Before_Bed': 'screen_time_before_bed',
    'Phone_Checks_Per_Day': 'phone_checks_per_day',
    'Apps_Used_Daily': 'apps_used_daily',
    'Time_on_Social_Media': 'time_on_social_media',
    'Time_on_Gaming': 'time_on_gaming',
    'Time_on_Education': 'time_on_education',
    'Phone_Usage_Purpose': 'phone_usage_purpose',
    'Family_Communication': 'family_communication',
    'Weekend_Usage_Hours': 'weekend_usage_hours',
    'Addiction_Level': 'addiction_level'
}, inplace=True)
df.head()

,ID,age,gender,school_grade,daily_usage_hours,sleep_hours,academic_performance,social_interactions,exercise_hours,anxiety_level,...,screen_time_before_bed,phone_checks_per_day,apps_used_daily,time_on_social_media,time_on_gaming,time_on_education,phone_usage_purpose,family_communication,weekend_usage_hours,addiction_level
0,1,13,Female,9th,4.0,6.1,78,5,0.1,10,...,1.4,86,19,3.6,1.7,1.2,Browsing,4,8.7,10.0
1,2,17,Female,7th,5.5,6.5,70,5,0.0,3,...,0.9,96,9,1.1,4.0,1.8,Browsing,2,5.3,10.0
2,3,13,Other,11th,5.8,5.5,93,8,0.8,2,...,0.5,137,8,0.3,1.5,0.4,Education,6,5.7,9.2
3,4,18,Female,12th,3.1,3.9,78,8,1.6,9,...,1.4,128,7,3.1,1.6,0.8,Social Media,8,3.0,9.8
4,5,14,Other,9th,2.5,6.7,56,4,1.1,1,...,1.0,96,20,2.6,0.9,1.1,Gaming,10,3.7,8.6


One naming note to remember for the analysis notebook: the first dataset has depression_label, a 0 or 1 flag. This dataset has depression_level, a 1 to 10 scale. Similar names, different meaning. Not combining these two fields directly.

In [18]:
pd.options.display.float_format = '{:.2f}'.format
df.head()

,ID,age,gender,school_grade,daily_usage_hours,sleep_hours,academic_performance,social_interactions,exercise_hours,anxiety_level,...,screen_time_before_bed,phone_checks_per_day,apps_used_daily,time_on_social_media,time_on_gaming,time_on_education,phone_usage_purpose,family_communication,weekend_usage_hours,addiction_level
0,1,13,Female,9th,4.00,6.10,78,5,0.10,10,...,1.40,86,19,3.60,1.70,1.20,Browsing,4,8.70,10.00
1,2,17,Female,7th,5.50,6.50,70,5,0.00,3,...,0.90,96,9,1.10,4.00,1.80,Browsing,2,5.30,10.00
2,3,13,Other,11th,5.80,5.50,93,8,0.80,2,...,0.50,137,8,0.30,1.50,0.40,Education,6,5.70,9.20
3,4,18,Female,12th,3.10,3.90,78,8,1.60,9,...,1.40,128,7,3.10,1.60,0.80,Social Media,8,3.00,9.80
4,5,14,Other,9th,2.50,6.70,56,4,1.10,1,...,1.00,96,20,2.60,0.90,1.10,Gaming,10,3.70,8.60


Set float display to two decimal places, same formatting as the first cleanup notebook.

Building user_id from the ID column with a D2 prefix, same convention used for the first dataset's D1 prefix.

In [19]:
df.rename(columns={'ID': 'user_id'}, inplace=True)
df['user_id'] = 'D2_' + df['user_id'].astype(str)
df.head()

,user_id,age,gender,school_grade,daily_usage_hours,sleep_hours,academic_performance,social_interactions,exercise_hours,anxiety_level,...,screen_time_before_bed,phone_checks_per_day,apps_used_daily,time_on_social_media,time_on_gaming,time_on_education,phone_usage_purpose,family_communication,weekend_usage_hours,addiction_level
0,D2_1,13,Female,9th,4.00,6.10,78,5,0.10,10,...,1.40,86,19,3.60,1.70,1.20,Browsing,4,8.70,10.00
1,D2_2,17,Female,7th,5.50,6.50,70,5,0.00,3,...,0.90,96,9,1.10,4.00,1.80,Browsing,2,5.30,10.00
2,D2_3,13,Other,11th,5.80,5.50,93,8,0.80,2,...,0.50,137,8,0.30,1.50,0.40,Education,6,5.70,9.20
3,D2_4,18,Female,12th,3.10,3.90,78,8,1.60,9,...,1.40,128,7,3.10,1.60,0.80,Social Media,8,3.00,9.80
4,D2_5,14,Other,9th,2.50,6.70,56,4,1.10,1,...,1.00,96,20,2.60,0.90,1.10,Gaming,10,3.70,8.60


In [20]:
df['user_id'].is_unique

True

user_id stays unique after adding the prefix.

In [21]:
df.shape

(3000, 23)

In [22]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 23 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   user_id                 3000 non-null   str    
 1   age                     3000 non-null   int64  
 2   gender                  3000 non-null   str    
 3   school_grade            3000 non-null   str    
 4   daily_usage_hours       3000 non-null   float64
 5   sleep_hours             3000 non-null   float64
 6   academic_performance    3000 non-null   int64  
 7   social_interactions     3000 non-null   int64  
 8   exercise_hours          3000 non-null   float64
 9   anxiety_level           3000 non-null   int64  
 10  depression_level        3000 non-null   int64  
 11  self_esteem             3000 non-null   int64  
 12  parental_control        3000 non-null   int64  
 13  screen_time_before_bed  3000 non-null   float64
 14  phone_checks_per_day    3000 non-null   int64  
 15

3000 rows, 23 columns after dropping Name and Location and adding user_id in place of the raw ID. No nulls, no duplicates.

In [23]:
df.to_csv('../Data/cleaned_teen_phone_addiction_dataset.csv', index=False)

Saved the cleaned data to a new CSV. Original file left unchanged. Both cleaned files come together next in Student_MH_Database_SQL.ipynb.